postprocessing/resstockpostproc/process_bsb_results.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+"""
+Script to take in raw BuildStockBatch results_csv / parquet and convert them to pub_annual version.
+
+Example usage:
+uv run resstockpostproc/process_bsb_results.py /path/to/bsb_raw_results /path/to/output_dir
+
+Note: bsb_raw_results folder must contain both baseline and upgrade files. Baseline file should be named
+results_up00.parquet and upgrade files should be named results_upXX.parquet where XX is the upgrade number. The can
+either be in their own folders (baseline and upgrades) or all be in the same folder.
+"""
+
+import sys
+import polars as pl
+from pathlib import Path
+from resstockpostproc.process_metadata import publish_baseline_annual_results, publish_upgrade_annual_results
+import re
+
+def process_results(raw_results_dir: str, output_dir: str) -> None:
+    output_path = Path(output_dir)
+    output_path.mkdir(parents=True, exist_ok=True)
+    result_files = list(Path(raw_results_dir).glob("*"))
+    baseline_files = [f for f in result_files if "up00" in f.name.lower()]
+    upgrade_files = [f for f in result_files if "up00" not in f.name.lower()]
+
+    if not baseline_files:
+        print("Error: No baseline or upgrade files found")
+        sys.exit(1)
+    if len(baseline_files) > 1:
+        print("Error: More than one baseline file found")
+        sys.exit(1)
+
+    baseline_file = baseline_files[0]
+    print(f"Processing baseline file: {baseline_file}")
+    baseline_df = pl.scan_parquet(baseline_file)
+    failed_bldgs = (
+        baseline_df.filter(pl.col("completed_status") == "Fail")
```

Comment:
Does ResStock ever have situations where the status is blank or "Success" but there are no numbers in the results data? ComStock has weird failure modes sometimes that we have to catch. If you haven't seen that probably don't need to worry about it.

Reply:
Yeah, those three are the only values I have seen. For the GHP upgrade with lots of failures (and possible weird corner cases), that's all we have.

<img width="1216" height="300" alt="image" src="https://github.com/user-attachments/assets/2ae4b37a-ec6e-478c-9104-d4eb9a075068" />